empress_gui.py

```diff
@@ -4,362 +4,819 @@
+                    self.parasite_tree_info = tk.Label(self.input_info_frame, text="Parasite/symbiont: "+os.path.basename(self.parasite_file_path)+": "+str(parasite_tree_tips_number)+" tips")
+                    self.parasite_tree_info.grid(row=1, column=0, sticky="w")
+                else: 
+                    messagebox.showinfo("Warning", "The input file cannot be read.")          
+            else:
+                messagebox.showinfo("Warning", "Please load a '.nwk' file.")
+            
+            self.load_files_var.set("Load files")
+
+        # Clicking on "Load mapping file" 
+        elif self.load_files_var.get() == "Load mapping file":
+            # initialdir is set to be the current working directory
+            input_file = tk.filedialog.askopenfilename(initialdir=os.getcwd(), title="Select a mapping file")
+            if Path(input_file).suffix == '.mapping': 
+                # Try to read in mapping file
+                self.recon_input.read_mapping(input_file)
+                self.mapping_file_path = None
+                if self.recon_input.tip_mapping is not None:
+                    self.mapping_file_path = input_file
+                    # Reset every time user successfully loads in a new mapping file
+                    self.reset("Load mapping file")
+                    self.mapping_info = tk.Label(self.input_info_frame, text="Mapping: "+os.path.basename(self.mapping_file_path))
+                    self.mapping_info.grid(row=2, column=0, sticky="w")
+
+                    # Enables the next step, setting DTL costs
+                    if self.recon_input.is_complete(): 
+                        self.view_tanglegram_btn.configure(state=tk.NORMAL)
+                        self.view_cost_space_btn.configure(state=tk.NORMAL)
+                        self.compute_reconciliations_btn.configure(state=tk.NORMAL)
+                        self.dtl_cost()
+                else: 
+                    messagebox.showinfo("Warning", "The input file cannot be read.")          
+            else:
+                messagebox.showinfo("Warning", "Please load a '.mapping' file.")
+            
+            self.load_files_var.set("Load files")
+
+    def compute_tree_tips(self, tree_type):
+        """Compute the number of tips for the host tree and parasite tree inputs."""
+        if tree_type == "host tree":
+            host_tree_object = dict_to_tree(self.recon_input.host_dict, tree.TreeType.HOST)
+            return len(host_tree_object.leaf_list())
+        elif tree_type == "parasite tree":
+            parasite_tree_object = dict_to_tree(self.recon_input.parasite_dict, tree.TreeType.PARASITE)
+            return len(parasite_tree_object.leaf_list())
+
+    def plot_cost_regions(self):
+        """Plot the cost regions using matplotlib and embed the graph in a tkinter window."""
+        self.view_solution_space_dropdown.configure(state=tk.DISABLED)
```

Comment:
You might want to consider moving 364-366 to a new function. Seems like the functionality is shared by other parts of the code.

Reply:
Done.